app/src/main/res/layout/fragment_favorites_list.xml

```diff
@@ -1,12 +1,28 @@
 <?xml version="1.0" encoding="utf-8"?>
 <LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
     xmlns:tools="http://schemas.android.com/tools"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
     android:background="@drawable/favorites_background"
     android:orientation="vertical"
     tools:context=".changes.ChangeListFragment">
 
+    <androidx.constraintlayout.widget.ConstraintLayout
```

Comment:
Did you try to remove the `ConstraintLayout` and just add the `ProgressBar`?

Reply:
This combination allows us to more finely control the size of the ProgressBar, otherwise the progress bar has to fill the entire screen in order to hide the content behind it. This way the ConstraintLayout is the actual overlay, while the Progress Bar only takes up a small amount of screen space

Reply:
I see. If the `ConstraintLayout` I would prefer to replace the outer layout with a `ConstraintLayout` and position all views within it. That would avoid the unneeded nesting.

Reply:
I can try to do that. That Constraint layout is to overlay. So I'll probably not be able to replace that

Reply:
You can have multiple views on top of each other in a ConstraintLayout. Once you show/hide them it should be fine.

Reply:
@Chase22 If I can help with that let me know.